### PR TITLE
Feature user profile

### PR DESCRIPTION
**Pull request by OlgaMardvilko**
Team work Olga Mardvilko and Ivan Runets under the leadership of Nikita Rudy.
The student profile page is created by the layout https://sonejka.github.io/rs-kittens/dist/admin-student-profile.html
Task User Profile (Lead team: Nikita Rudy).
Description of the task https://github.com/rolling-scopes/rsschool-docs/blob/master/tasks/user-profile.md.
To view the page, test routers / student

### CODE REVIEW

**Comment by dzmitry-varabei** [comment]
Hello! Is this PR ready for review? Or are you going to refactor something?

File: src/app/components/ProfileUser/Avatar.tsx [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+type Props = {
+    githubId: string;
+};
+
+const Avatar = (props: Props) => {
+    const { githubId } = props;
+    return (
+        <React.Fragment>
+            <p className="mb-2">
+                <img src={getGithubAvatar(githubId)} alt={`${githubId}`} className={cn('avatar')} />
```
Comment (Shastel) [inline src/app/components/ProfileUser/Avatar.tsx]: Wrapping into template string is redundant here

File: src/app/components/ProfileForm/parsers-formatters.ts [hunk 2]
```diff
@@ -9,3 +9,5 @@ export const emailFormatter = (emails: Array<{ primary: boolean; value: string }
 export const emailParser = (email: string) => {
     return [{ primary: true, value: email }];
 };
+
+export const required = (value: string) => (value ? undefined : 'Required');
```
Comment (Shastel) [inline src/app/components/ProfileForm/parsers-formatters.ts]: shall it be not just for strings?

File: src/app/components/ProfileUser/Contacts.tsx [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    contacts: Contacts;
+};
+
+const Contacts = (props: Props) => (
+    <React.Fragment>
+        <h4 className="badge badge-info">Contact information</h4>
+
+        {Object.keys(props.contacts).map(contact => {
+            return (
+                <React.Fragment key={contact.toString()}>
```
Comment (Shastel) [inline src/app/components/ProfileUser/Contacts.tsx]: Could you please explain the reason to create Fragment here?

File: src/app/components/ProfileUser/MainMentor.tsx [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+                    <p>Experience</p>
+                    <h3>{yearsInFrontEnd}</h3>
+                </Col>
+                <Col sm="3">
+                    <p>City</p>
+                    <h3>{city}</h3>
+                </Col>
+                <Col sm="3">
+                    <p>
+                        <a href={`https://github.com/${githubName}`}>{`${props.githubName}`}</a>
```
Comment (Shastel) [inline src/app/components/ProfileUser/MainMentor.tsx]: Don't see a reason to wrap githubName into templatestring

File: src/app/components/ProfileUser/MainStudent.tsx [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+            </Col>
+            <Col sm="3">
+                <p>English Level</p>
+                <h3>{`${props.englishLevel}`}</h3>
+            </Col>
+            <Col sm="3">
+                <p>City</p>
+                <h3>{`${props.city}`}</h3>
+            </Col>
+            <Col sm="3">
```
Comment (Shastel) [inline src/app/components/ProfileUser/MainStudent.tsx]: ctrl+c / ctrl + v?

File: src/app/core/api/mentor.ts [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+        {
+            _id: 'dadajndjksandjk3n4324',
+            isActive: true,
+            courseId: 'rs-course-2018-1',
+            role: 'mentor',
+        },
+    ],
+};
+
+export const getMentorProfile = Promise.resolve(mentor);
```
Comment (Shastel) [inline src/app/core/api/mentor.ts]: Shall you convert it to function that returns promise?

File: src/app/core/reducers/user.ts [hunk 7]
```diff
@@ -31,6 +31,7 @@ export function userReducer(state = initialState, action: Action<any>): UserStat
             };
         }
         case USER.PROFILE_UPDATE_OK: {
+            // console.log(action.payload);
```
Comment (Shastel) [inline src/app/core/reducers/user.ts]: Could you please remove this log?

File: src/app/core/store.ts [hunk 8]
```diff
@@ -6,14 +6,15 @@ import IHotModule from './models/hot-module.model';
...
+import { composeWithDevTools } from 'redux-devtools-extension';
 
 declare const module: IHotModule;
 
 const configureStore = (history: History) => {
     const routerMiddleware = createRouterMiddleware(history);
     // const sagaMiddleware = createSagaMiddleware();
 
-    const store = createStore(rootReducer, applyMiddleware(routerMiddleware, thunk));
+    const store = createStore(rootReducer, composeWithDevTools(applyMiddleware(routerMiddleware, thunk)));
```
Comment (Shastel) [inline src/app/core/store.ts]: I guess it should be enable only in dev mode

File: src/app/components/ProfileUser/Contacts.tsx (hunk 3 above)
Comment (OlgaMardvilko) [reply src/app/components/ProfileUser/Contacts.tsx]: it's changed
For now it will always render 5 rows
In this case, Fragment was  to add checks and render rows in which contact information is provided. There  are used if user didn't provide some contact information, for example telegram or skype missed.

File: src/app/components/ProfileUser/MainMentor.tsx (hunk 4 above)
Comment (OlgaMardvilko) [reply src/app/components/ProfileUser/MainMentor.tsx]: it's changed

File: src/app/components/ProfileForm/parsers-formatters.ts (hunk 2 above)
Comment (OlgaMardvilko) [reply src/app/components/ProfileForm/parsers-formatters.ts]: it's changed

File: src/app/components/ProfileUser/Avatar.tsx (hunk 1 above)
Comment (OlgaMardvilko) [reply src/app/components/ProfileUser/Avatar.tsx]: it's changed

File: src/app/components/ProfileUser/MainStudent.tsx (hunk 5 above)
Comment (OlgaMardvilko) [reply src/app/components/ProfileUser/MainStudent.tsx]: it's changed

File: src/app/core/api/mentor.ts (hunk 6 above)
Comment (OlgaMardvilko) [reply src/app/core/api/mentor.ts]: We did not understand what the function is for here

File: src/app/core/reducers/user.ts (hunk 7 above)
Comment (OlgaMardvilko) [reply src/app/core/reducers/user.ts]: it's changed

File: src/app/core/store.ts (hunk 8 above)
Comment (OlgaMardvilko) [reply src/app/core/store.ts]: it's changed

**Comment by OlgaMardvilko** [comment]
@Shastel Hello! We have made changes in accordance with your comments. Please see our PR. More comments can be found in the description of each issue.

File: src/app/components/ProfileUser/Avatar.tsx [hunk 9]
```diff
@@ -12,7 +12,7 @@ const Avatar = (props: Props) => {
     return (
         <React.Fragment>
             <p className="mb-2">
-                <img src={getGithubAvatar(githubId)} alt={`${githubId}`} className={cn('avatar')} />
+                <img src={getGithubAvatar(githubId)} alt={githubId} className={cn('avatar')} />
```
Comment (Shastel) [inline src/app/components/ProfileUser/Avatar.tsx]: Could you please also remove fragment around here?

File: src/app/components/ProfileUser/MainStudent.tsx [hunk 10]
```diff
@@ -10,35 +10,38 @@ type Props = {
...
-                    <a href={`https://github.com/${props.githubName}`}>
-                        <FontAwesomeIcon icon={faGithub} />
-                    </a>
-                </h3>
-            </Col>
-        </Row>
-    </React.Fragment>
-);
+const MainInformation = (props: Props) => {
+    const { englishLevel, city, githubName } = props;
```
Comment (Shastel) [inline src/app/components/ProfileUser/MainStudent.tsx]: Shall we use destructuring in function declaration?

File: src/app/components/ProfileForm/MentoringForm.tsx [hunk 11]
```diff
@@ -0,0 +1,78 @@
...
+import * as React from 'react';
+import { FormGroup, Label } from 'reactstrap';
+import { Field } from 'redux-form';
+import { required } from './parsers-formatters';
+import { QUANTITY } from '../../reference-data';
+
+class MentoringForm extends React.PureComponent<any, any> {
+    render() {
+        return [
+            <div key="1" className="row">
```
Comment (Shastel) [inline src/app/components/ProfileForm/MentoringForm.tsx]: Could you please name keys somehow logical instead of incrementing it?

File: src/app/components/ProfileUser/Education.tsx [hunk 12]
```diff
@@ -0,0 +1,47 @@
...
+    isInternshipNeeded: boolean;
+    isWorkNeeded: boolean;
+};
+
+const Education = (props: Props) => (
+    <React.Fragment>
+        <h2 className="badge badge-info">Education</h2>
+        {Object.keys(props.primaryEducation).map(edication => {
+            return (
+                <React.Fragment key={edication.toString()}>
```
Comment (Shastel) [inline src/app/components/ProfileUser/Education.tsx]: Fragment is redundant here

File: src/app/components/ProfileUser/Education.tsx [hunk 13]
```diff
@@ -0,0 +1,47 @@
...
+type Props = {
+    primaryEducation: PrimaryEducation;
+    isInternshipNeeded: boolean;
+    isWorkNeeded: boolean;
+};
+
+const Education = (props: Props) => (
+    <React.Fragment>
+        <h2 className="badge badge-info">Education</h2>
+        {Object.keys(props.primaryEducation).map(edication => {
```
Comment (Shastel) [inline src/app/components/ProfileUser/Education.tsx]: Did you mean `education`?

File: src/app/components/ProfileUser/MainMentor.tsx [hunk 14]
```diff
@@ -0,0 +1,46 @@
...
+                </Col>
+                <Col sm="3">
+                    <p>Experience</p>
+                    <h3>{yearsInFrontEnd}</h3>
+                </Col>
+                <Col sm="3">
+                    <p>City</p>
+                    <h3>{city}</h3>
+                </Col>
+                <Col sm="3">
```
Comment (Shastel) [inline src/app/components/ProfileUser/MainMentor.tsx]: I think i saw same component somewhere around

File: src/app/components/ProfileUser/MentorProfile.tsx [hunk 15]
```diff
@@ -0,0 +1,112 @@
...
+
+                            <Other notes={props.mentorProfile.notes} />
+                        </Col>
+                    </Row>
+                </Jumbotron>
+            </main>
+            <div className="text-right">
+                <Button color="danger" onClick={props.handleReject}>
+                    <FontAwesomeIcon icon={faBan} /> Reject
+                </Button>{' '}
```
Comment (Shastel) [inline src/app/components/ProfileUser/MentorProfile.tsx]: Why did you need `{' '}`

File: src/app/components/ProfileUser/Mentors.tsx [hunk 16]
```diff
@@ -0,0 +1,23 @@
...
+};
+
+const Mentors = (props: Props) => {
+    const { mentors } = props;
+    return (
+        <React.Fragment>
+            {mentors
+                ? mentors.map(mentor => (
+                      <p key={mentor._id}>
+                          Mentor: <a href="#">{mentor._id}</a>
```
Comment (Shastel) [inline src/app/components/ProfileUser/Mentors.tsx]: Shall it be `mentor.name` or smth?

File: src/app/components/ProfileUser/Mentors.tsx [hunk 17]
```diff
@@ -0,0 +1,23 @@
...
+type Props = {
+    mentors: Array<IUserBase | IUser> | undefined;
+};
+
+const Mentors = (props: Props) => {
+    const { mentors } = props;
+    return (
+        <React.Fragment>
+            {mentors
+                ? mentors.map(mentor => (
```
Comment (Shastel) [inline src/app/components/ProfileUser/Mentors.tsx]:
```javascript
mentors && ...
```

File: src/app/core/api/mentor.ts (hunk 6 above)
Comment (Shastel) [reply src/app/core/api/mentor.ts]: Because basically you have to refetch smth on demand, not only once, so it have to be a function.

File: src/app/core/selectors/index.ts [hunk 18]
```diff
@@ -1,3 +1,26 @@
+import { RootState } from '../reducers';
+
+export const getUserProfile = (state: RootState) => {
+    if (state.user && state.user.profile) {
+        return state.user.profile;
+    }
+    return;
```
Comment (Shastel) [inline src/app/core/selectors/index.ts]: `return` statement is redundant here

File: src/app/components/ProfileUser/MentorProfile.tsx (hunk 15 above)
Comment (OlgaMardvilko) [reply src/app/components/ProfileUser/MentorProfile.tsx]: Because we needed to have an indent between the buttons

File: src/app/components/ProfileUser/Education.tsx (hunk 13 above)
Comment (OlgaMardvilko) [reply src/app/components/ProfileUser/Education.tsx]: This changed

File: src/app/components/ProfileUser/Education.tsx (hunk 12 above)
Comment (OlgaMardvilko) [reply src/app/components/ProfileUser/Education.tsx]: This changed

File: src/app/components/ProfileUser/Avatar.tsx (hunk 9 above)
Comment (OlgaMardvilko) [reply src/app/components/ProfileUser/Avatar.tsx]: This changed

File: src/app/core/selectors/index.ts (hunk 18 above)
Comment (OlgaMardvilko) [reply src/app/core/selectors/index.ts]: This changed

File: src/app/components/ProfileUser/Mentors.tsx (hunk 17 above)
Comment (OlgaMardvilko) [reply src/app/components/ProfileUser/Mentors.tsx]: This changed

File: src/app/core/api/mentor.ts (hunk 6 above)
Comment (OlgaMardvilko) [reply src/app/core/api/mentor.ts]: This changed

File: src/app/components/ProfileForm/MentoringForm.tsx (hunk 11 above)
Comment (OlgaMardvilko) [reply src/app/components/ProfileForm/MentoringForm.tsx]: This changed

File: src/app/components/ProfileUser/MentorProfile.tsx (hunk 15 above)
Comment (Shastel) [reply src/app/components/ProfileUser/MentorProfile.tsx]: and why you can't do it by css?

File: src/app/components/ProfileForm/ContactsForm.tsx [hunk 19]
```diff
@@ -2,21 +2,13 @@ import ReduxFormInput from 'components/ReduxFormInput';
...
-                    <Label>City</Label>
-                    <Field name="city" type="select" component={ReduxFormInput}>
-                        <option value="">(Empty)</option>
-                        {CITIES.map(city => <option key={city.id}>{city.name}</option>)}
-                    </Field>
-                </FormGroup>
-                <FormGroup className="col-md-6">
+            <div key="contacts_phone" className="row">
+                <FormGroup className="col-md-12">
                     <Label className="field-required">Phone (in the form 8-XXX-XXX-XX-XX)</Label>
```
Comment (Shastel) [inline src/app/components/ProfileForm/ContactsForm.tsx]: Shall we also change phone number to international format?
cc @dzmitry-varabei

File: src/app/components/ProfileUser/PagePrint.tsx [hunk 20]
```diff
@@ -0,0 +1,21 @@
...
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faPrint } from '@fortawesome/free-solid-svg-icons';
+
+type Props = {
+    namePage: string;
+};
+
+const PagePrint = (props: Props) => (
+    <React.Fragment>
+        <h2>{props.namePage}</h2>
```
[the remark this answers is not in the document]
Comment (OlgaMardvilko) [reply src/app/components/ProfileUser/PagePrint.tsx]: This changed

File: src/app/components/ProfileForm/ContactsForm.tsx (hunk 19 above)
Comment (dzmitry-varabei) [reply src/app/components/ProfileForm/ContactsForm.tsx]: @Shastel sure! 
@OlgaMardvilko please go ahead

Comment (OlgaMardvilko) [reply src/app/components/ProfileForm/ContactsForm.tsx]: This changed

File: src/app/components/ProfileUser/MainMentor.tsx (hunk 14 above)
Comment (OlgaMardvilko) [reply src/app/components/ProfileUser/MainMentor.tsx]: This changed

File: src/app/containers/ProfileUser/index.tsx [hunk 21]
```diff
@@ -0,0 +1,44 @@
...
+class UserProfile extends React.Component<Props, any> {
+    render() {
+        if (this.props.role === 'mentor') {
+            return this.props.profile ? (
+                <Mentor mentorProfile={this.props.profile} handleApprove={() => {}} handleReject={() => {}} />
+            ) : (
+                <h2>loading...</h2>
+            );
+        }
+        return this.props.profile ? (
```
[the remark this answers is not in the document]
Comment (OlgaMardvilko) [reply src/app/containers/ProfileUser/index.tsx]: This changed

**Comment by OlgaMardvilko** [comment]
@Shastel Hello! Please look again. Make changes to what you showed. Thank you.

File: src/app/components/ProfileUser/Mentors.tsx [hunk 22]
```diff
@@ -0,0 +1,22 @@
+import * as React from 'react';
+import { IUser, IUserBase } from '../../core/models';
+
+type Props = {
+    mentors: Array<IUserBase | IUser> | undefined;
+};
+
+const Mentors = (props: Props) => {
+    const { mentors } = props;
+    return (
```
Comment (Shastel) [inline src/app/components/ProfileUser/Mentors.tsx]: Let's use early return here

File: src/app/components/ProfileUser/ModalExpel.tsx [hunk 23]
```diff
@@ -0,0 +1,76 @@
...
+                    <Form onSubmit={handleSubmit}>
+                        <ModalHeader toggle={this.toggle}>STUDENT EXPULSION</ModalHeader>
+                        <ModalBody>
+                            <p>Do you really want to expel {`${firstName} ${lastName}`}</p>
+                            <FormGroup>
+                                <Field
+                                    name="expelReason"
+                                    label="Please, explain the reason"
+                                    component={ReduxFormInput}
+                                    required={true}
```
[the remark this answers is not in the document]
Comment (Shastel) [reply src/app/components/ProfileUser/ModalExpel.tsx]: I'm sure that following: `< Field required />;` will be working for you

File: src/app/components/ProfileUser/Other.tsx [hunk 24]
```diff
@@ -0,0 +1,20 @@
...
+    notes: string;
+};
+
+const Other = (props: Props) => (
+    <React.Fragment>
+        <h2 className="badge badge-info">Other</h2>
+        <Row>
+            <Col md="4">Additional Information:</Col>
+            <Col md="8">
+                <p>{`${props.notes}`}</p>
```
Comment (Shastel) [inline src/app/components/ProfileUser/Other.tsx]: Extra template string literal
